dynamic/info.go

```diff
@@ -83,6 +83,22 @@ func providerInfo(ctx context.Context, p run.Provider, value parameterize.Value)
 			}
 		},
 	}
+	// Add presumed best-effort GitHub org to the provider info.
+	// We do not set the GitHubOrg field for a local dynamic provider.
+	if value.Remote != nil {
+		// https://github.com/opentofu/registry/issues/1337:
+		// Due to discrepancies in the registry protocol/implementation,
+		// we infer the Terraform provider's source code repository via the following assumptions:
+		// - The provider's source code is hosted at github.com
+		// - The provider's github org, for providers, is the namespace field of the registry name
+		// Example:
+		//
+		// opentofu.org/provider/hashicorp/random -> "hashicorp" is deduced to be the github org.
+		// Note that this will only work for the provider (not the module) protocol.
+		urlFields := strings.Split(value.Remote.URL, "/")
+		ghOrg := urlFields[len(urlFields)-2]
+		prov.GitHubOrg = ghOrg
```

Comment:
We should set `prov.Repository`.
```suggestion
		urlFields := strings.Split(value.Remote.URL, "/")
		prov.GitHubOrg = urlFields[len(urlFields)-2]
		prov.Repository = fmt.Sprintf("https://github.com/%s/terraform-provider-%s",
			prov.GitHubOrg, urlFields[len(urlFields)-1])
```

Reply:
I wonder if there's a better way to differentiate between "this provider has pulumi source code" and "this provider only has terraform source code".
While the [docs are pretty clear that the field refers to the source code repo](https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfbridge/info/info.go#L70), for Pulumi-maintained providers this field is expected to be a Pulumi provider repository as opposed to a Terraform one.
And elsewhere does [express this difference](https://github.com/pulumi/pulumi-terraform-bridge/blob/2a8b877dbe500f33980524c68855f0e732c3d6aa/pkg/tfgen/generate_schema.go#L588).